sa/storage-authority.go

```diff
@@ -673,6 +674,117 @@ func (ssa *SQLStorageAuthority) FinalizeAuthorization(authz core.Authorization)
 	return
 }
 
+var authorizationTables = []string{
+	"authz",
+	"pendingAuthorizations",
+}
+
+const getAuthorizationIDsMax = 1000
+
+func getAuthorizationIDsByDomain(db *gorp.DbMap, ident string, now time.Time) ([]string, error) {
+	var allIDs []string
+	for _, tableName := range authorizationTables {
+		_, err := db.Select(
+			&allIDs,
+			fmt.Sprintf(
+				`SELECT id FROM %s
+         WHERE identifier = :ident AND
+         status != :invalid AND
+         status != :revoked AND
+         expires > :now
+         LIMIT :limit`,
+				tableName,
+			),
+			map[string]interface{}{
+				"ident":   ident,
+				"invalid": string(core.StatusInvalid),
+				"revoked": string(core.StatusRevoked),
+				"now":     now,
+				"limit":   getAuthorizationIDsMax,
+			},
+		)
+		if err != nil {
+			return nil, err
+		}
+		if len(allIDs) == getAuthorizationIDsMax {
+			break
+		}
+	}
+	return allIDs, nil
+}
+
+func revokeAuthorizations(db *gorp.DbMap, authIDs []string) (int64, int64, error) {
+	results := []int64{0, 0}
+	quotedIDs := []string{}
+	for _, id := range authIDs {
+		quotedIDs = append(quotedIDs, strconv.Quote(id))
```

Comment:
This isn't enough. It also has to escape newlines, commas and other characters.

Instead, use the built-in escaping. It costs a bit more code but you can make the question marked template string with 

    func QMarks(num int) string {
      if num == 0 {
        return "()"
      }
      return "(?" + strings.Repeat(",?", num-1) + ")"
    }

And then you have to make a `[]interface{}` for the args and append all of the current args into  before passing it to `db.Exec(queryStr, args...)` in order to get the typing right.